src/service/cid-impl.js

```diff
@@ -118,13 +124,64 @@ export class Cid {
         'The CID scope and cookie name must only use the characters ' +
         '[a-zA-Z0-9-_.]+\nInstead found: %s',
         getCidStruct.scope);
-    return consent.then(() => {
-      return viewerForDoc(this.ampdoc).whenFirstVisible();
-    }).then(() => {
-      return getExternalCid(this, getCidStruct,
-          opt_persistenceConsent || consent);
+    return isOptedOutOfCid(this.ampdoc).then(optedOut => {
+      // TODO(aghassemi): null or '' ?
+      if (optedOut) {
+        return null;
```

Comment:
done, also moved the check after consent check (since user can optout while consent is pending)